src/wp-includes/post.php

```diff
@@ -3743,14 +3743,21 @@ function wp_post_mime_type_where( $post_mime_types, $table_alias = '' ) {
  * @see wp_delete_attachment()
  * @see wp_trash_post()
  *
- * @param int  $post_id      Optional. Post ID. Default 0.
+ * @param int  $post_id      Post ID.
  * @param bool $force_delete Optional. Whether to bypass Trash and force deletion.
  *                           Default false.
  * @return WP_Post|false|null Post data on success, false or null on failure.
  */
-function wp_delete_post( $post_id = 0, $force_delete = false ) {
+function wp_delete_post( $post_id, $force_delete = false ) {
```

Comment:
I think this should be left unchanged. Why? Because a fatal error could occur otherwise. The `_doing_it_wrong()` will alert the dev that they aren't supplying the required parameter.

```suggestion
function wp_delete_post( $post_id = 0, $force_delete = false ) {
```